lib/rubocop/cop/rails/orm_abstraction.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # Avoid breaking ORM abstraction.
+      # @example
+      #
+      #   # bad
+      #   where(foo_id: foo.id)
+      #
+      #   # bad
+      #   create(foo_id: foo.id)
+      #
+      #   # good
+      #   where(foo: foo)
+      #
+      #   # good
+      #   create(foo: foo)
+
+      class ORMAbstraction < Cop
+        def_node_matcher :bad_pair?, <<-PATTERN
+          (pair {str_type? sym_type? dstr_type?} (send !nil? {:id :uuid}))
+        PATTERN
+
+        def on_pair(node)
+          return unless bad_pair?(node)
+          return if !node.key.source.match(BAD_RXP)
+
+          corrector = Corrector.new(processed_source.buffer)
```

Comment:
It would be cool if I could re-use this corrector for the autocorrect, but it is too late at night right now to figure that out.

Reply:
Yeah, doesn't look possible